### PR TITLE
On-Prem System Requirements

### PR DESCRIPTION
**Pull request by resmaa**
Sub Page added to on-Premises Installation 
  System Requirements
    1. Quick Start Version
    2. Enterprise Version

### CODE REVIEW

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

Approval requirements bypassed by manually added approval.

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/355#" title="Author self-approved">resmaa</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

Approval requirements bypassed by manually added approval.

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/355#" title="Author self-approved">resmaa</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by spectro-prow** [comment]
Hi @resmaa. Thanks for your PR.

I'm waiting for a [spectrocloud](https://github.com/orgs/spectrocloud/people) member to verify that this patch is reasonable to test. If it is, they should reply with `/ok-to-test` on its own line. Until that is done, I will not automatically test new commits in this PR, but the usual testing commands by org members will still work. Regular contributors should [join the org](https://github.com/orgs/spectrocloud/people) to skip this step.

Once the patch is verified, the new status will be reflected by the `ok-to-test` label.

I understand the commands that are listed [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: packages/docs/content/12-enterprise-version/00-on-prem-system-requirements.md [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+
+* Datacenter
+   - IPs for application workload services (e.g.: LoadBalancer services).
+   - Subnet with egress access to the internet (direct or via proxy):
+   - For proxy: HTTP_PROXY, HTTPS_PROXY (both required).
+   - NTP configured on all Hosts.
+   - Shared Storage between vSphere hosts.
+   - VMware vCenter permissions.
+
+* Zone Tagging helps in storage allocation while dynamic storage volume provisioning is in place. Use vSphere tags to label zones in your vSphere environment. The task assumes that your vCenter Server environment includes three clusters, cluster1, cluster2, and cluster3, with the node VMs on all three clusters. In the task, you create two tag categories, k8s-zone and k8s-region. You tag the clusters as three zones, zone-a, zone-b, and zone-c, and mark the data center as a region, region-1.
```
Comment (arvind-sys) [inline packages/docs/content/12-enterprise-version/00-on-prem-system-requirements.md:66]: Can add Screenshots?

File: packages/docs/content/12-enterprise-version/00-on-prem-system-requirements.md [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+
+### vSphere Environment  Prerequisites
+
+* Datacenter
+   - IPs for application workload services (e.g.: LoadBalancer services).
+   - Subnet with egress access to the internet (direct or via proxy):
+   - For proxy: HTTP_PROXY, HTTPS_PROXY (both required).
+   - NTP configured on all Hosts.
+   - Shared Storage between vSphere hosts.
+   - VMware vCenter permissions.
```
Comment (arvind-sys) [inline packages/docs/content/12-enterprise-version/00-on-prem-system-requirements.md:64]: Same with table is listed below. So might be remove this?

File: packages/docs/content/12-enterprise-version/00-on-prem-system-requirements.md [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+| | vApp instance configuration
+| vSphere Tagging| Create vSphere Tag
+| | Edit vSphere Tag
+
+
+### Optional Settings 
+The requirement configurations given below are optional.
+
+* DNS Mapping
+   - Static IP DNS: [Name servers] Comma separated DNS addresses (e.g: 8.8.8.8, 192.168.0.8), required only for static IP allocation.
```
Comment (arvind-sys) [inline packages/docs/content/12-enterprise-version/00-on-prem-system-requirements.md:175]: No. DNS Mapping is providing a proper URL by which Spectro Cloud Console can be accessed. 
Can be mentioned something like:
An URL mapped to internal/external DNS server which can be used to access Spectro Cloud Console

File: packages/docs/content/12-enterprise-version/00-on-prem-system-requirements.md [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+* SMTP Settings
+   - Configure SMTP settings to enable the Spectro Cloud platform to send out email notifications. Email Notifications are sent out to new users when they are on boarded to the platform so they can activate their accounts.
```
Comment (arvind-sys) [inline packages/docs/content/12-enterprise-version/00-on-prem-system-requirements.md:177]: Also for Password resets and Alert notifications too.

File: packages/docs/content/12-enterprise-version/00-on-prem-system-requirements.md [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+* Certificate (optional)
```
Comment (arvind-sys) [inline packages/docs/content/12-enterprise-version/00-on-prem-system-requirements.md:178]: Nope.

Provide SSL certificate to access Spectro Cloud Console in a secured https way.

File: packages/docs/content/12-enterprise-version/00-on-prem-system-requirements.md [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+* FTP Location (backups)
+   - An accessible FTP location to backup data.
+   	
+# Quick Start Mode
+
+The Spectro Cloud On-Prem Quick Start Mode is a single node installation of the Spectro Cloud platform. This deployment is used to quickly understand the capabilities of the Spectro Cloud platform. For production purposes the enterprise version is recommended. The minimum requirements for Quick Start version is outlined below:
+
+### Hardware Requirements 
+
+Hardware requirements will describe the CPU, memory, and disk requirements for the nodes.
```
Comment (arvind-sys) [inline packages/docs/content/12-enterprise-version/00-on-prem-system-requirements.md:189]: Can mention minimum requirement. And also we can mention that the installer will take care of allocating the resources.

File: packages/docs/content/12-enterprise-version/00-on-prem-system-requirements.md [hunk 7]
```diff
@@ -0,0 +1,284 @@
...
+* 4 virtual CPUs 
+* Secondary storage of 80 GB distributed across multiple HDDs.
+
+### Network Requirements
+
+* Quick Start will operate on a single instance which requires connectivity through a proxy or non-proxy environment. 
+* This instance can be configured with static IP addresses or dynamically allocated IP addresses.
+
+<InfoBox>
+ Make sure that your Datacenter CIDR IP address does not overlap Kubernetes PodCIDR range. Kubernetes PodCIDR range settings can be changed while installing Quick Start Version
```
Comment (arvind-sys) [inline packages/docs/content/12-enterprise-version/00-on-prem-system-requirements.md:200]: No Quick Start Version

can be 
while installing Spectro Cloud On-Prem Quick Start Application

File: packages/docs/content/12-enterprise-version/00-on-prem-system-requirements.md [hunk 8]
```diff
@@ -0,0 +1,284 @@
...
+### Platform Requirements
+
+* Ubuntu Operating System
+* Both of the on-prem modes support vSphere version 6.7 and above.
+
+
+### Common Network Requirements
+
+* The instances of on-prem deployment requires connectivity through a proxy or non-proxy environment. 
+* This instance can be configured with static IP addresses or dynamically allocated IP addresses.
```
Comment (arvind-sys) [inline packages/docs/content/12-enterprise-version/00-on-prem-system-requirements.md:37]: Will not be required here. As it differs for QS and enterprise.

File: packages/docs/content/12-enterprise-version/00-on-prem-system-requirements.md [hunk 9]
```diff
@@ -0,0 +1,284 @@
...
+### Hardware Requirements 
+
+Hardware requirements will describe the CPU, memory, and disk requirements for the nodes.
+* 8GB memory
+* 4 virtual CPUs 
+* Secondary storage of 80 GB distributed across multiple HDDs.
+
+### Network Requirements
+
+* Quick Start will operate on a single instance which requires connectivity through a proxy or non-proxy environment.
```
Comment (arvind-sys) [inline packages/docs/content/12-enterprise-version/00-on-prem-system-requirements.md:196]: The instances should be reachable to the whitelisted domains either through Proxy or Without Proxy Network Connections.

File: packages/docs/content/12-enterprise-version/00-on-prem-system-requirements.md [hunk 10]
```diff
@@ -0,0 +1,284 @@
...
+<InfoBox>
+ Make sure that your Datacenter CIDR IP address does not overlap Kubernetes PodCIDR range. Kubernetes PodCIDR range settings can be changed while installing Quick Start Version
+</InfoBox>
+
+# Enterprise Mode
+
+The Spectro Cloud On-Premise Enterprise version is a highly available multinode infrastructure for production purposes. The minimum number of nodes/VMs required by this environment is 3. Depending on the number of nodes the requirements changes. The minimum requirements are outlined below:
+
+### Hardware Requirements 
+Hardware requirements will describe the CPU, memory, and disk requirements for the nodes.
```
Comment (arvind-sys) [inline packages/docs/content/12-enterprise-version/00-on-prem-system-requirements.md:208]: Mention minimum required

File: packages/docs/content/12-enterprise-version/00-on-prem-system-requirements.md [hunk 11]
```diff
@@ -0,0 +1,284 @@
...
+* Each VM requires 8GB memory resulting in 24GB RAM in total .
+* 4 vCPUs per node * 3 nodes requiring 12 vCPUs
+* Secondary storage of 240GB distributed across multiple HDDs.
+
+### Network Requirements
+
+* A Range or a Subnet of at least 5 static IP addresses is required for the installation and ongoing management. 
+* The infrastructure will require a proxy/non proxy environment for accessibility.
```
Comment (arvind-sys) [inline packages/docs/content/12-enterprise-version/00-on-prem-system-requirements.md:216]: The instances should be reachable to the whitelisted domains either through Proxy or Without Proxy Network Connections.

File: packages/docs/content/12-enterprise-version/00-on-prem-system-requirements.md [hunk 12]
```diff
@@ -0,0 +1,284 @@
...
+Hardware requirements will describe the CPU, memory, and disk requirements for the nodes.
+* Each VM requires 8GB memory resulting in 24GB RAM in total .
+* 4 vCPUs per node * 3 nodes requiring 12 vCPUs
+* Secondary storage of 240GB distributed across multiple HDDs.
+
+### Network Requirements
+
+* A Range or a Subnet of at least 5 static IP addresses is required for the installation and ongoing management. 
+* The infrastructure will require a proxy/non proxy environment for accessibility. 
+* For high availability purposes, you may choose to distribute the three VMs across multiple compute clusters.
```
Comment (arvind-sys) [inline packages/docs/content/12-enterprise-version/00-on-prem-system-requirements.md:217]: In Bracket or as a note can mention the tagging part

File: packages/docs/content/12-enterprise-version/00-on-prem-system-requirements.md [hunk 13]
```diff
@@ -0,0 +1,284 @@
...
+
+<InfoBox>
+ Make sure that your Datacenter CIDR IP address does not overlap Kubernetes PodCIDR range. Kubernetes PodCIDR range settings can be changed while installing Enterprise version
+</InfoBox>
+
+
+## Private Cloud Gateway Installer
+
+A Private Cloud Gateway is a Spectro Cloud component that enables the communication between Spectro Cloud's management console and a VMware based private data center. The gateway needs to be installed by the users in their VMware environments using a private cloud gateway installer appliance.
```
Comment (arvind-sys) [inline packages/docs/content/12-enterprise-version/00-on-prem-system-requirements.md:226]: As this On prem we can mention:

A Private Cloud Gateway is a Spectro Cloud component that enables the communication between Spectro Cloud's management console and a data center which is not accessible directly via the Datacenter hosting Spectro Cloud On-Prem Application. The gateway needs to be installed by the users in their VMware environments using a private cloud gateway installer OVA. For more details <link to Gateway installer section>

File: packages/docs/content/12-enterprise-version/00-on-prem-system-requirements.md [hunk 14]
```diff
@@ -0,0 +1,284 @@
...
+### Hardware Requirement 
+
+* The infrastructure will require a proxy/non proxy environment for outgoing traffic. 
+* [Whitelist](#proxywhitelists) lists the proxy requirements for enabling the Spectro Cloud management console.
+* The instance can be configured with static IP addresses or dynamically allocated IP addresses (DHCP or Static).
+* vSphere Environment  Pre-requisites
+     - IPs for application workload services (e.g.: LoadBalancer services).
+     - Subnet with egress access to the internet (direct or via proxy):
+     - For proxy: HTTP_PROXY, HTTPS_PROXY (both required).
+     - NTP configured on all Hosts.
```
Comment (arvind-sys) [inline packages/docs/content/12-enterprise-version/00-on-prem-system-requirements.md:237]: esxi

**Review by jgautam — APPROVED** [review]
Looks Good. Need to review in the docs-latest site for better readability , so approving it.

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

Approval requirements bypassed by manually added approval.

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/355#pullrequestreview-658131132" title="Approved">jgautam</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/355#" title="Author self-approved">resmaa</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**